opal/mca/memory/patcher/memory_patcher_component.c

```diff
@@ -0,0 +1,432 @@
+ *                         reserved.
+ * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2009-2014 Cisco Systems, Inc.  All rights reserved.
+ * Copyright (c) 2013-2016 Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2016      Research Organization for Information Science
+ *                         and Technology (RIST). All rights reserved.
+ * ******** Add IBM COPYRIGHT HERE ***********
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "memory_patcher.h"
+
+#include "opal/util/output.h"
+#include "opal/util/show_help.h"
+#include "opal/mca/memory/base/empty.h"
+#include "opal/mca/memory/base/base.h"
+#include "opal/memoryhooks/memory.h"
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <errno.h>
+#include <stdarg.h>
+#include <sys/mman.h>
+#include <dlfcn.h>
+#include <assert.h>
+#include <sys/time.h>
+#include <sys/syscall.h>
+
+#include "memory_patcher.h"
+#undef opal_memory_changed
+
+static int patcher_open(void);
+static int patcher_close(void);
+static int patcher_register(void);
+static int patcher_query (int *);
+
+static int mca_memory_patcher_priority;
+
+opal_memory_patcher_component_t mca_memory_patcher_component = {
+    .super = {
+        .memoryc_version = {
+            OPAL_MEMORY_BASE_VERSION_2_0_0,
+
+            /* Component name and version */
+            .mca_component_name = "patcher",
+            MCA_BASE_MAKE_VERSION(component, OPAL_MAJOR_VERSION, OPAL_MINOR_VERSION,
+                                  OPAL_RELEASE_VERSION),
+
+            /* Component open and close functions */
+            .mca_open_component = patcher_open,
+            .mca_close_component = patcher_close,
+            .mca_register_component_params = patcher_register,
+        },
+        .memoryc_data = {
+            /* The component is checkpoint ready */
+            MCA_BASE_METADATA_PARAM_CHECKPOINT
+        },
+
+        /* Memory framework functions. */
+        .memoryc_query = patcher_query,
+        .memoryc_register = opal_memory_base_component_register_empty,
+        .memoryc_deregister = opal_memory_base_component_deregister_empty,
+        .memoryc_set_alignment = opal_memory_base_component_set_alignment_empty,
+    },
+
+    /* Component-specific data, filled in later (compiler will 0/NULL
+       it out) */
+};
+
+#if OPAL_MEMORY_PATCHER_HAVE___SYSCALL_PROTO && OPAL_MEMORY_PATCHER_HAVE___SYSCALL
+/* calling __syscall is preferred on some systems when some arguments may be 64-bit. it also
+ * has the benefit of having an off_t return type */
+#define memory_patcher_syscall __syscall
+#else
+#define memory_patcher_syscall syscall
+#endif
+
+#if 0
```

Comment:
Can you at least put a good explanation in a comment about this?  I.e., what this commented out code is, why it's commented out, why it's not deleted, ...etc.?